pkg/syncer/mysqlcluster/statefullset.go

```diff
@@ -0,0 +1,662 @@
+	})
+
+	fileMode := int32(0644)
+	in[1] = ensureVolume(in[1], confMapVolumeName, core.VolumeSource{
+		ConfigMap: &core.ConfigMapVolumeSource{
+			LocalObjectReference: core.LocalObjectReference{
+				Name: s.cluster.GetNameForResource(api.ConfigMap),
+			},
+			DefaultMode: &fileMode,
+		},
+	})
+
+	in[2] = ensureVolume(in[2], dataVolumeName, core.VolumeSource{
+		PersistentVolumeClaim: &core.PersistentVolumeClaimVolumeSource{
+			ClaimName: dataVolumeName,
+		},
+	})
+
+	return in
+}
+
+func (s *sfsSyncer) ensureVolumeClaimTemplates(in []core.PersistentVolumeClaim) []core.PersistentVolumeClaim {
+	initPvc := false
+	if len(in) == 0 {
+		in = make([]core.PersistentVolumeClaim, 1)
+		initPvc = true
+	}
+	data := in[0]
+
+	data.Name = dataVolumeName
+
+	if initPvc {
+		// This can be set only when creating new PVC. It ensures that PVC can be
+		// terminated after deleting parent MySQL cluster
+		data.ObjectMeta.OwnerReferences = s.getOwnerReferences()
+	}
+
+	data.Spec = s.cluster.Spec.VolumeSpec.PersistentVolumeClaimSpec
+
+	in[0] = data
+
+	return in
+}
+
+func (s *sfsSyncer) getEnvSourcesFor(name string) []core.EnvFromSource {
+	envSources := []core.EnvFromSource{}
+	if name == containerCloneName && len(s.cluster.Spec.InitBucketSecretName) > 0 {
+		envSources = append(envSources, core.EnvFromSource{
+			SecretRef: &core.SecretEnvSource{
+				LocalObjectReference: core.LocalObjectReference{
+					Name: s.cluster.Spec.InitBucketSecretName,
+				},
+			},
+		})
+	}
+	if name == containerHelperName {
+		envSources = append(envSources, core.EnvFromSource{
+			Prefix: "MYSQL_",
+			SecretRef: &core.SecretEnvSource{
+				LocalObjectReference: core.LocalObjectReference{
+					Name: s.cluster.Spec.SecretName,
+				},
+			},
+		})
+	}
+	return envSources
+}
+
+func (s *sfsSyncer) getVolumeMountsFor(name string) []core.VolumeMount {
+	switch name {
+	case containerInitName:
+		return []core.VolumeMount{
+			core.VolumeMount{
+				Name:      confVolumeName,
+				MountPath: ConfVolumeMountPath,
+			},
+			core.VolumeMount{
+				Name:      confMapVolumeName,
+				MountPath: ConfMapVolumeMountPath,
+			},
+		}
+
+	case containerCloneName, containerMysqlName, containerHelperName:
+		return []core.VolumeMount{
+			core.VolumeMount{
+				Name:      confVolumeName,
+				MountPath: ConfVolumeMountPath,
+			},
+			core.VolumeMount{
+				Name:      dataVolumeName,
+				MountPath: DataVolumeMountPath,
+			},
+		}
+
+	case containerHeartBeatName, containerKillerName:
+		return []core.VolumeMount{
+			core.VolumeMount{
+				Name:      confVolumeName,
+				MountPath: ConfVolumeMountPath,
+			},
+		}
+	}
+
+	return nil
+}
+
+func ensureVolume(in core.Volume, name string, source core.VolumeSource) core.Volume {
+	in.Name = name
+	in.VolumeSource = source
+
+	return in
+}
+
+func (s *sfsSyncer) getOwnerReferences(ors ...[]metav1.OwnerReference) []metav1.OwnerReference {
+
+	rs := []metav1.OwnerReference{
+		s.cluster.GetOwnerReferences()[0],
```

Comment:
`GetOwnerReferences()` returns a list of owners references  of the cluster.  But a cluster does not have owner reference. You should create a owner reference as I did in `pkg/apis/mysqlv1alpha1/mysqlcluster_util.go`

```go
// AsOwnerReference returns the MysqlCluster owner references.
func (c *MysqlCluster) AsOwnerReference() metav1.OwnerReference {
	trueVar := true
	return metav1.OwnerReference{
		APIVersion: SchemeGroupVersion.String(),
		Kind:       "MysqlCluster",
		Name:       c.Name,
		UID:        c.UID,
		Controller: &trueVar,
	}
}
```

You can create a function that returns the owner reference on `* sfsSyncer` object.